lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -132,8 +132,12 @@ class HoistLayoutConversion : public OpRewritePattern<ConvertLayoutOp> {
         src->getDialect()->getTypeID() != TypeID::get<arith::ArithDialect>())
       return failure();
 
-    // TODO(jlebar): Why do we exclude these two ops specifically?
-    if (isa<arith::TruncIOp, arith::TruncFOp>(src))
+    // TODO(jlebar): Why do we exclude arith trunc ops specifically?
+    // arith.select needs to be excluded because dot_operand layout hoisting
+    // will go through the mask, which is i1 type. Mask parents are most likely
+    // of integer types. When lowering shared -> dot_operand layout, integer
+    // types are not supported. 
```

Comment:
can you replace this whole comment by something explaining that for dot_operand layout we don't support all the types and type conversions.